sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamWindowRel.java

```diff
@@ -0,0 +1,317 @@
+  public BeamWindowRel(
+      RelOptCluster cluster,
+      RelTraitSet traitSet,
+      RelNode input,
+      List<RexLiteral> constants,
+      RelDataType rowType,
+      List<Group> groups) {
+    super(cluster, traitSet, input, constants, rowType, groups);
+  }
+
+  @Override
+  public PTransform<PCollectionList<Row>, PCollection<Row>> buildPTransform() {
+    Schema outputSchema = CalciteUtils.toSchema(getRowType());
+    final List<FieldAggregation> analyticFields = Lists.newArrayList();
+    this.groups.stream()
+        .forEach(
+            anAnalyticGroup -> {
+              List<Integer> partitionKeysDef = anAnalyticGroup.keys.toList();
+              List<Integer> orderByKeys = Lists.newArrayList();
+              List<Boolean> orderByDirections = Lists.newArrayList();
+              List<Boolean> orderByNullDirections = Lists.newArrayList();
+              anAnalyticGroup.orderKeys.getFieldCollations().stream()
+                  .forEach(
+                      fc -> {
+                        orderByKeys.add(fc.getFieldIndex());
+                        orderByDirections.add(
+                            fc.direction == RelFieldCollation.Direction.ASCENDING);
+                        orderByNullDirections.add(
+                            fc.nullDirection == RelFieldCollation.NullDirection.FIRST);
+                      });
+              int lowerB = Integer.MAX_VALUE; // Unbounded by default
+              int upperB = Integer.MAX_VALUE; // Unbounded by default
+              if (anAnalyticGroup.lowerBound.isCurrentRow()) {
+                lowerB = 0;
+              } else if (anAnalyticGroup.lowerBound.isPreceding()) {
+                // pending
+              } else if (anAnalyticGroup.lowerBound.isFollowing()) {
+                // pending
+              }
+              if (anAnalyticGroup.upperBound.isCurrentRow()) {
+                upperB = 0;
+              } else if (anAnalyticGroup.upperBound.isPreceding()) {
+                // pending
+              } else if (anAnalyticGroup.upperBound.isFollowing()) {
+                // pending
+              }
+              final int lowerBFinal = lowerB;
+              final int upperBFinal = upperB;
+              List<AggregateCall> aggregateCalls = anAnalyticGroup.getAggregateCalls(this);
+              aggregateCalls.stream()
+                  .forEach(
+                      anAggCall -> {
+                        List<Integer> argList = anAggCall.getArgList();
+                        Schema.Field field =
+                            CalciteUtils.toField(anAggCall.getName(), anAggCall.getType());
+                        Combine.CombineFn combineFn =
+                            AggregationCombineFnAdapter.createCombineFn(
+                                anAggCall, field, anAggCall.getAggregation().getName());
+                        FieldAggregation fieldAggregation =
+                            new FieldAggregation(
```

Comment:
Even though you will stop executions on some unsupported cases, I think it is still ok to keep this class definition (e.g. no need to remove those unused field).